compiler/ras/ILValidator.hpp

```diff
@@ -19,187 +19,51 @@
+   bool validate();
 
-      TR::Node *anyLiveNode()
-         {
-         Iterator iter(*this);
-         return iter.currentNode();
-         }
-
-      void startNewWindow()
-         {
-         TR_ASSERT(_liveOffsets.isEmpty(), "Cannot close LiveNodeWindow when there are still live nodes");
-         _basis = _sideTable.size();
-         }
-
-      public:
-
-      class Iterator
-         {
-         LiveNodeWindow      &_window;
-         TR_BitVectorIterator _bvi;
-         int32_t              _currentOffset; // -1 means iteration is past end
-
-         bool isPastEnd(){ return _currentOffset < 0; }
-
-         public:
-
-         Iterator(LiveNodeWindow &window):_window(window),_bvi(window._liveOffsets),_currentOffset(0xdead /* anything non-negative */)
-            {
-            stepForward();
-            }
-
-         TR::Node *currentNode()
-            {
-            if (isPastEnd())
-               return NULL;
-            else
-               return _window._sideTable.getAt(_window._basis + _currentOffset)._node;
-            }
-
-         void stepForward()
-            {
-            TR_ASSERT(!isPastEnd(), "Can't stepForward a LiveNodeWindow::Iterator that's already past end");
-            if (_bvi.hasMoreElements())
-               _currentOffset = _bvi.getNextElement();
-            else
-               _currentOffset = -1;
-            }
-
-         public: // operators
-
-         void operator ++() { stepForward(); }
-         };
-      };
-
-   NodeSideTable<NodeState>  _nodeStates;
-   LiveNodeWindow            _liveNodes;
-
-   bool isValidSoFar(){ return _isValidSoFar; }
-   void validityRule(Location &location, bool condition, const char *formatStr, ...);
-
-   protected: // Individual validation routines
-
-   // These return false to indicate a validity error was found.
-   // Note that these are given verb names ("validateXxx") instead of adjective
-   // names ("xxxIsValid") because generally they are stateful.
-
-   // Called on every node, before all other validation routines
-   //
-   virtual void validateNode(Location &location);
-
-   //
-   // Additional validation hooks
-   //
-
-   virtual void validateEndOfExtendedBlock(Location &location);
+   private:
+   TR::Compilation  *_comp;
+   TR::Compilation *comp();
 
-   protected:
+   // The Validation rules are categorized into three types based on the
+   // "scope" required to validate them.
+   // For example, a MethodValidationRule would require information about
+   // the entire Method(encapsulated as a TR::ResolvedMethodSymbol) in order
+   // to find out if the IL satisifies the given condition.
+   // Whereas a NodeValidationRule checks whether a particular TR::Node has
+   // some property. Meaning a NodeValidationRule doesn't need to keep track
+   // of already encountered nodes or peek into other blocks to see whether
+   // a particular Node is valid or not.
+   std::vector<TR::MethodValidationRule *> _methodValidationRules;
```

Comment:
I'd change these comments to split them to a per-member basis, and you should use Doxygen style throughout.